types/prosemirror-model/prosemirror-model-tests.ts

```diff
@@ -14,6 +21,9 @@ domOutputSpec = ['div', ['div', 0]];
 domOutputSpec = ['div', ['div', { class: 'foo' }]];
 domOutputSpec = ['div', ['div', { class: 'foo' }, 0]];
 
+domOutputSpec = {dom: divNode};
+domOutputSpec = {dom: divNode, contentDOM: parentDivNode};
+
 export const nodeSpec: model.NodeSpec = {
```

Comment:
I think it would be helpful to add tests for `ParseRule.consuming`. This can be accomplished using `$ExpectType` and `$ExpectError`.

Link: https://github.com/microsoft/dtslint

Reply:
Okay. How extensive should the tests be? An `$ExpectError` for every basic type (string, object, array) that is not allowed in ParseRule.consuming? Or just that it is optional and that it takes booelan?

Reply:
Unless I'm testing to convince myself one type is assignable to another, I usually go with `$ExpectType` to check something has the right type:

```
const rule = {} as ParseRule
// $ExpectType boolean | null | undefined
rule.consuming
```